### PR TITLE
Implement map drawing and movement

### LINKED ISSUE #3: Implement Map Drawing Routines
The map draw routines draw the overworld map, on each action

### PR DESCRIPTION
**Pull request by 6167656e74323431**
Adds map drawing routines and adds some stricture for coordination between graphical components.

Fixes #3

### CODE REVIEW

**Review by sbrad044 — DISMISSED** [review]

**Review by siddharthgowda — APPROVED** [review]
Everything looks good. The only thing is that for InventoryInteractor the package name is use_case instead of usecase. I think that's the reason for the conflict.

File: src/main/java/com/mg105/user_interface/MainMenu.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.mg105.user_interface;
+
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import org.jetbrains.annotations.NotNull;
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/user_interface/MainMenu.java:7]: Good use for javafx for implementing the UI and visualizing the MainMenu.

File: src/main/java/com/mg105/user_interface/MapDrawer.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        );
+        isVisible = false;
+
+        tiles = new HashMap<>(6);
+        tiles.put(TileType.FLOOR, new Image(RoomUpdater.class.getResourceAsStream("/tiles/floor.png")));
+        tiles.put(TileType.WALL, new Image(RoomUpdater.class.getResourceAsStream("/tiles/wall.png")));
+        tiles.put(TileType.EXIT, new Image(RoomUpdater.class.getResourceAsStream("/tiles/exit.png")));
+        tiles.put(TileType.CHEST, new Image(RoomUpdater.class.getResourceAsStream("/tiles/chest.png")));
+        tiles.put(TileType.PLAYER, new Image(RoomUpdater.class.getResourceAsStream("/tiles/player.png")));
+        tiles.put(TileType.OPPONENT_SET, new Image(RoomUpdater.class.getResourceAsStream("/tiles/battle.png")));
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/user_interface/MapDrawer.java:50]: good idea for implementing and storing the tiles by HashMap and visualize it by importing images.

**Review by hoodiexxx — APPROVED** [review]

**Review by zhaoan12 — DISMISSED** [review]
Looks good

**Review by sbrad044 — APPROVED** [review]

**Review by zhaoan12 — APPROVED** [review]

**Review by siddharthgowda — APPROVED** [review]
Looks good and nice documentation

**Pull request merged**